test/systemtests/swarm_test.go

```diff
@@ -0,0 +1,561 @@
+package systemtests
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type swarm struct {
+	node *node
+	env  string
+}
+
+func (s *systemtestSuite) NewSwarmExec(n *node) *swarm {
+	w := new(swarm)
+	w.node = n
+	w.env = s.basicInfo.SwarmEnv + " "
+	return w
+}
+
+func (w *swarm) newContainer(node *node, containerID, name string) (*container, error) {
+	cont := &container{node: node, containerID: containerID, name: name}
+
+	out, err := node.exec.getIPAddr(cont, "eth0")
+	if err != nil {
+		return nil, err
+	}
+	cont.eth0.ip = out
+
+	out, err = cont.node.exec.getIPv6Addr(cont, "eth0")
+	if err == nil {
+		cont.eth0.ipv6 = out
+	}
+
+	return cont, nil
+}
+
+func (w *swarm) runContainer(spec containerSpec) (*container, error) {
+	var namestr, netstr, dnsStr, labelstr string
+	if spec.networkName != "" {
+		netstr = spec.networkName
+
+		if spec.serviceName != "" {
+			netstr = spec.serviceName
+		}
+		if spec.tenantName != "" && spec.tenantName != "default" {
+			netstr = netstr + "/" + spec.tenantName
+		}
+
+		netstr = "--net=" + netstr
+	}
+
+	if spec.imageName == "" {
+		spec.imageName = "alpine"
+	}
+
+	if spec.commandName == "" {
+		spec.commandName = "sleep 60m"
+	}
+
+	if spec.name != "" {
+		namestr = "--name=" + spec.name
+	}
+
+	if spec.dnsServer != "" {
+		dnsStr = "--dns=" + spec.dnsServer
+	}
+
+	if len(spec.labels) > 0 {
+		l := "--label="
+		for _, label := range spec.labels {
+			labelstr += l + label + " "
+		}
+	}
+	logrus.Infof("Starting a container running %q on %s", spec.commandName, w.node.Name())
```

Comment:
similar to k8. It might be good to maintain the node where the container is scheduled on. some use cases for k8s.  like for ping test you can show container on which node is trying to ping. This log is printed in the currents tests. But this will be very useful info for debugging